### PR TITLE
gadgets: Add README.md for image-based gadgets

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
Add README.md for image-based gadgets. This should be used by artifacthub to show the documentation for individual gadgets as well! 

Related https://github.com/inspektor-gadget/inspektor-gadget/issues/2727

This PR only adds documentation for following gadgets:

- snapshot_process
- snapshot_socket
- trace_exec
- trace_open

### CODE REVIEW

**Comment by mqasimsarfraz** [comment]
I am putting this on hold till https://github.com/inspektor-gadget/inspektor-gadget/issues/2736 is fixed

File: gadgets/snapshot_process/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+title: 'Using snapshot process'
+weight: 20
+description: >
+  Gather information about running processes.
+---
```
Comment (alban) [inline gadgets/snapshot_process/README.md:6]: I think we can remove the hugo headers since artifacthub does not use them.

File: gadgets/snapshot_process/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+> ⚠️ This feature is experimental and could change without prior notification. Check the installation guide to enable [experimental features](../../docs/getting-started/install-linux.md#experimental-features).
```
Comment (alban) [inline gadgets/snapshot_process/README.md:8]: Since this doc aims to be published on artifacthub and all gadgets published on artifacthub are necessarily image-based gadgets, I suggest to remove that line: this image-based gadget is not more experimental than other image-based gadgets.

What do you think?

**Review by alban — COMMENTED** [review]
Thanks for starting this documentation!

File: gadgets/snapshot_process/README.md (hunk 2 above)
Comment (mqasimsarfraz) [reply gadgets/snapshot_process/README.md:8]: > What do you think?

The main reason to include it here is to ensure the user doesn't get any surprises with `kubectl gadget run snapshot_process` like they need to deploy Inspektor Gadget with experimental to be able to follow steps on this guide?

Comment (mqasimsarfraz) [reply gadgets/snapshot_process/README.md:1]: Or perhaps we should rephrase it to say `This feature requires Inspektor Gadget has been installed with experimental features enabled`

**Comment by mqasimsarfraz** [comment]
> I am putting this on hold till https://github.com/inspektor-gadget/inspektor-gadget/issues/2736 is fixed

I will resume work this PR!

**Comment by alban** [comment]
> I will resume work this PR!

Thanks! Once all gadgets have a README, we can request that artifacthub.io marks them as "official" (it is last requirement currently missing for requesting the official status). 

https://artifacthub.io/docs/topics/repositories/#official-status

**Comment by alban** [comment]
I think the README.md files should have additional sections:
- ebpf parameters
- fields

They could be automatically updated from gadget.yaml. And if we have a tool to keep it up to date, that tool should be made available for third party gadgets. Do you have plans how to do this? This is a bit similar to the tool in #2586.

But that should be done in a future PR: it is better to have a minimalist README.md as soon as possible, so we can request  the official status to artifacthub.io before implementing everything.

**Comment by mqasimsarfraz** [comment]
Fixed by: 
- https://github.com/inspektor-gadget/inspektor-gadget/pull/3279
- https://github.com/inspektor-gadget/inspektor-gadget/pull/3358
